src/controllers/todo-controller.ts

```diff
@@ -0,0 +1,54 @@
+import lodash from 'lodash';
+
+import { BaseController } from './base-controller';
+import { NextFunction, Response, Router } from 'express';
+import { Validation } from '@helpers';
+import { Todo } from '@models';
+import {
+  AppContext,
+  Errors,
+  ExtendedRequest,
+  ValidationFailure,
+} from '@typings';
+import {
+  createTodoValidator
+} from '@validators';
+
+export class TodoController extends BaseController {
+  public basePath: string = '/todos';
+  public router: Router = Router();
+
+  constructor(ctx: AppContext) {
+    super(ctx);
+    this.initializeRoutes();
+  }
+
+  private initializeRoutes() {
+    this.router.post(
+      `${this.basePath}`,
+      createTodoValidator(this.appContext),
+      this.createTodo,
+    );
+  }
+
+  private createTodo = async (req:ExtendedRequest, res: Response, next: NextFunction) => {
+    const failures: ValidationFailure[] = Validation.extractValidationErrors(
```

Comment:
This is not ideal in my opinion even though I understand that's how boilerplate does it.

I think the validation should be handled by repositories instead of controller layer. This way, in case, we build RPC endpoints or GraphQL endpoints etc, the validation logic can be shared.

Controller should act as simple proxy.